framework/src/base/Assembly.C

```diff
@@ -1522,6 +1524,19 @@ Assembly::addCachedJacobianContributions(SparseMatrix<Number> & jacobian)
 }
 
 void
+Assembly::registerUserObjectShapeVariable(unsigned int var)
+{
+  std::vector<unsigned int> & reg = _user_object_shape_variables;
+
+  // add variable number to the list of registered variables
+  reg.push_back(var);
+
+  // sort and keep elements unique
+  std::sort(reg.begin(), reg.end());
+  reg.erase(std::unique(reg.begin(), reg.end()), reg.end());
```

Comment:
Because it is written to only at the start of the simulation but iterated over a LOT. A vector send to be the more appropriate days structure.